### ISSUE TITLE
Compile time error for Struct with isFabricScoped attribute, when Struct is used as command argument type

### ISSUE THREAD

**Issue by lazarkov** [project-chip/zap#1255]
For the specific command I was not able to compile the code when the struct had isFabricScopet set to true

**Struct**
```
<struct name="MessageStruct" isFabricScoped="true">
    <cluster code="0x0097"/>
    <item fieldId="0" name="MessageID" type="octet_string" length="16" optional="false"/>
</struct>
```

**Command**
```
<command source="client" code="0x00" name="PresentMessagesRequest" isFabricScoped="true" optional="false">
  <description>
    Command for requesting messages be presented
  </description>
  <arg name="Messages" type="MessageStruct" array="true" optional="false"/>
</command>
```

This will lead to compile time error:
[complete-error-log.txt](https://github.com/project-chip/zap/files/14057926/complete-error-log.txt)

**Comment by bzbarsky-apple** [comment]
@gmarcosb @lazarkov This is not a ZAP issue.  The Matter SDK/codegen just does not support this and it needs to be fixed there.

Note that the spec technically does not allow this either, so the spec needs to be fixed too.

**Comment by gmarcosb** [comment]
Yeah I noticed that it looks like it's the `.zapt` template & not a zap issue; I just figured as there was a reported issue I'd track here & we can figure out where to fix (I'll probably take a look at it when I get around to it)

**Comment by brdandu** [comment]
can we close this issue and open one in the CHIP? If you need help with the .zapt templates, I can help you with that if you let me know what you are looking to generate.

**Comment by gmarcosb** [comment]
Sorry fix is in template @ https://github.com/project-chip/connectedhomeip/pull/37969
